app/uk/gov/hmrc/agentmapping/controller/MappingController.scala

```diff
@@ -174,71 +133,37 @@ class MappingController @Inject()(
       }
   }
 
-  def createPreSubscriptionMapping(utr: Utr) = Action.async { implicit request =>
-    implicit val hc = fromHeadersAndSession(request.headers, None)
-    authorised(AuthProviders(GovernmentGateway) and AffinityGroup.Agent)
-      .retrieve(Retrievals.credentials and Retrievals.agentCode and Retrievals.allEnrolments) {
-        case credentials ~ agentCodeOpt ~ allEnrolments =>
-          captureIdentifiersAndAgentCodeFrom(allEnrolments, agentCodeOpt) match {
-            case Some(identifiers) =>
-              identifiers
-                .map(identifier => createMappingInRepository(utr, identifier, credentials.providerId))
-                .reduce((f1, f2) => f1.flatMap(b1 => f2.map(b2 => b1 & b2)))
-                .map(
-                  allConflicted =>
-                    if (allConflicted)
-                      Conflict
-                    else
-                    Created)
-            case None => Future.successful(Forbidden)
-          }
-      }
-      .recover {
-        case ex: NoActiveSession =>
-          Logger(getClass).warn("No active session whilst trying to create mapping", ex)
-          Unauthorized
-        case ex: AuthorisationException =>
-          Logger(getClass).warn("Authorisation exception whilst trying to create mapping", ex)
-          Forbidden
-      }
-  }
-
-  def deletePreSubscriptionMapping(utr: Utr) = Action.async { implicit request =>
+  def createPreSubscriptionMapping(utr: Utr) =
```

Comment:
It's now clear after this refactor, that the `createMapping` and `createPreSubscriptionMapping` functions are identical except for the Arn/Utr tax identifier passed to `createMappingInRepository`. Perhaps extract a common endpoint function for them both?